library/src/main/scala/scala/scalajs/runtime/StackTrace.scala

```diff
@@ -153,9 +153,9 @@ object StackTrace {
    *  display the function name.
    */
   private def extractClassMethod(functionName: String): (String, String) = {
-    val PatC = """^ScalaJS\.c\.([^\.]+)(?:\.prototype)?\.([^\.]+)$""".re
-    val PatI = """^(?:Object\.)?ScalaJS\.i\.((?:_[^_]|[^_])+)__([^\.]+)$""".re
-    val PatM = """^(?:Object\.)?ScalaJS\.m\.([^.\.]+)$""".re
+    val PatC = """^(?:ScalaJS\.c\.|\$c_)([^\.]+)(?:\.prototype)?\.([^\.]+)$""".re
+    val PatI = """^(?:Object\.)?(?:ScalaJS\.s\.|\$s_)((?:_[^_]|[^_])+)__([^\.]+)$""".re
+    val PatM = """^(?:Object\.)?(?:ScalaJS\.m\.|\$m_)([^.\.]+)$""".re
```

Comment:
The problem is that VMs vary very much in the amount of information they put in their traces. So we cannot always recover things that we can verify. For example, on Rhino we don't ever get anything for `className` and `methodName`.

StackTrace is all *best effort*, but it doesn't guarantee anything because the VMs don't guarantee anything. Hence, untestable. :-(